testnet/stacks-node/src/run_loop/nakamoto.rs

```diff
@@ -0,0 +1,1023 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+use std::sync::atomic::AtomicBool;
+use std::sync::mpsc::sync_channel;
+use std::sync::{Arc, Mutex};
+use std::thread::JoinHandle;
+use std::{cmp, thread};
+
+use stacks::burnchains::bitcoin::address::{BitcoinAddress, LegacyBitcoinAddressType};
+use stacks::burnchains::Burnchain;
+use stacks::chainstate::burn::db::sortdb::SortitionDB;
+use stacks::chainstate::burn::BlockSnapshot;
+use stacks::chainstate::coordinator::comm::{CoordinatorChannels, CoordinatorReceivers};
+use stacks::chainstate::coordinator::{
+    static_get_canonical_affirmation_map, static_get_heaviest_affirmation_map,
+    static_get_stacks_tip_affirmation_map, ChainsCoordinator, ChainsCoordinatorConfig,
+    CoordinatorCommunication,
+};
+use stacks::chainstate::stacks::db::{ChainStateBootData, StacksChainState};
+use stacks::chainstate::stacks::miner::{signal_mining_blocked, signal_mining_ready, MinerStatus};
+use stacks::core::StacksEpochId;
+use stacks::net::atlas::{AtlasConfig, AtlasDB, Attachment};
+use stacks_common::types::PublicKey;
+use stacks_common::util::hash::Hash160;
+use stacks_common::util::{get_epoch_time_secs, sleep_ms};
+use stx_genesis::GenesisData;
+
+use crate::burnchains::make_bitcoin_indexer;
+use crate::globals::Globals as GenericGlobals;
+use crate::monitoring::start_serving_monitoring_metrics;
+use crate::nakamoto_node::{self, StacksNode, BLOCK_PROCESSOR_STACK_SIZE, RELAYER_MAX_BUFFER};
+use crate::node::{
+    get_account_balances, get_account_lockups, get_names, get_namespaces,
+    use_test_genesis_chainstate,
+};
+use crate::run_loop::neon;
+use crate::run_loop::neon::Counters;
+use crate::syncctl::{PoxSyncWatchdog, PoxSyncWatchdogComms};
+use crate::{
+    run_loop, BitcoinRegtestController, BurnchainController, Config, EventDispatcher, Keychain,
+};
+
+pub const STDERR: i32 = 2;
+pub type Globals = GenericGlobals<nakamoto_node::relayer::RelayerDirective>;
+
+#[cfg(test)]
+const UNCONDITIONAL_CHAIN_LIVENESS_CHECK: u64 = 30;
+
+#[cfg(not(test))]
+const UNCONDITIONAL_CHAIN_LIVENESS_CHECK: u64 = 300;
+
+/// Coordinating a node running in nakamoto mode. This runloop operates very similarly to the neon runloop.
+pub struct RunLoop {
+    config: Config,
+    globals: Option<Globals>,
+    counters: Counters,
+    coordinator_channels: Option<(CoordinatorReceivers, CoordinatorChannels)>,
+    should_keep_running: Arc<AtomicBool>,
+    event_dispatcher: EventDispatcher,
+    #[allow(dead_code)]
+    pox_watchdog: Option<PoxSyncWatchdog>, // can't be instantiated until .start() is called
+    is_miner: Option<bool>,       // not known until .start() is called
+    burnchain: Option<Burnchain>, // not known until .start() is called
+    pox_watchdog_comms: PoxSyncWatchdogComms,
```

Comment:
Not necessary anymore

Reply:
Right now, I'm just using it as a placeholder for an `ibd` global, I think as we move towards `Argon`, there will need to be a new struct for this, but I'd prefer to hold off on making a decision on that for now.